src/main/scala/cz/alenkacz/db/postgresscala/Connection.scala

```diff
@@ -0,0 +1,13 @@
+package cz.alenkacz.db.postgresscala
+
+import scala.concurrent.Future
+
+trait Connection {
+    def isConnected: Boolean
+
+    def sendQuery[T](query: String, deserializer: Row => T): Future[Seq[T]]
+    def sendQuery[T](query: String): Future[Option[T]]
```

Comment:
It's strange to me that if I expect one row then I cannot have custom deserializer.